client/verta/verta/dataset/_dataset.py

```diff
@@ -53,3 +58,62 @@ def _path_component_to_repr_lines(path_component_msg):
             lines.append("    SHA-256 checksum: {}".format(path_component_msg.sha256))
 
         return lines
+
+    def _set_commit_and_blob_path(self, commit, blob_path):
+        """
+        Associate this blob with a commit and path to enable downloads.
+
+        Parameters
+        ----------
+        commit : :class:`verta._repository.commit.Commit`
+            Commit this blob was gotten from.
+        blob_path : str
+            Location of this blob within its Repository.
+
+        """
+        self._commit = commit
+        self._blob_path = blob_path
+
+    def download(self, component_path, filepath=None, chunk_size=32*(10**3)):
+        """
+        Downloads `component_path` from this dataset if ModelDB-managed versioning was enabled.
+
+        Parameters
+        ----------
+        component_path : str
+            Original path of the file in this dataset to download.
+        filepath : str, optional
+            Filepath to download `component_path` to. If not provided, the file will be downloaded
+            to the current directory.
+        chunk_size : int, default 32 kB
+            Number of bytes to download at a time.
+
+        """
+        if self._commit is None and self._blob_path is None:
+            raise RuntimeError(
+                "this dataset cannot be used for downloads;"
+                " consider using `commit.get()` to obtain a download-capable dataset"
+                " if ModelDB-managed versioning was enabled"
+            )
+        if filepath is None:
+            filepath = os.path.basename(component_path)
+
+        # backend will return error if `component_path` not found/versioned
+        url = self._commit._get_url_for_artifact(self._blob_path, component_path, "GET").url
+
+        # stream download to avoid overwhelming memory
+        response = _utils.make_request("GET", url, self._commit._conn, stream=True)
+        try:
+            _utils.raise_for_http_error(response)
+
+            # create parent dirs
+            pathlib2.Path(filepath).parent.mkdir(parents=True, exist_ok=True)  # pylint: disable=no-member
+
+            # read response stream into file
+            with open(filepath, 'wb') as f:
+                for chunk in response.iter_content(chunk_size=chunk_size):
+                    f.write(chunk)
```

Comment:
I'd recommend writing to a location that has a different name and then renaming. The use case is when people check a given location to see if the file is there and skip the download (that we might even implement ourselves). So people only get the final file if everything goes fine.

Reply:
Done